isisd/isis_cli.c

```diff
@@ -1436,6 +1436,50 @@ void cli_show_isis_srgb(struct vty *vty, struct lyd_node *dnode,
 		yang_dnode_get_string(dnode, "./upper-bound"));
 }
 
+/*
+ * XPath: /frr-isisd:isis/instance/segment-routing/srlb
+ */
+DEFPY (isis_sr_local_block_label_range,
+       isis_sr_local_block_label_range_cmd,
+       "segment-routing local-block (16-1048575)$lower_bound (16-1048575)$upper_bound",
+       SR_STR
+       "Segment Routing Local Block label range\n"
+       "The lower bound of SRLB (16-1048575)\n"
+       "The upper bound of SRLB (16-1048575)\n")
+{
+	nb_cli_enqueue_change(vty, "./segment-routing/srlb/lower-bound",
+			      NB_OP_MODIFY, lower_bound_str);
+	nb_cli_enqueue_change(vty, "./segment-routing/srlb/upper-bound",
+			      NB_OP_MODIFY, upper_bound_str);
+
+	return nb_cli_apply_changes(vty, NULL);
+}
+
+DEFPY (no_isis_sr_local_block_label_range,
+       no_isis_sr_local_block_label_range_cmd,
+       "no segment-routing local-block [(16-1048575) (16-1048575)]",
+       NO_STR
+       SR_STR
+       "Segment Routing Local Block label range\n"
+       "The lower bound of SRLB (16-1048575)\n"
+       "The upper bound of SRLB (block size may not exceed 65535)\n")
```

Comment:
Is the size of the block being checked someplace to make certain the user didn't do something dumb? :-) I'll look for this later in the code, just wanted to verify here while it was "in my mind."

Reply:
Value for SRLB (and for SRGB) are checked due to yang data model rules:
```        container srlb {
          description
            "Local blocks to be advertised.";
          must "./upper-bound > ./lower-bound";
          leaf lower-bound {
              type uint32;
              default "15000";
              description
                "Lower value in the label range.";
          }
          leaf upper-bound {
              type uint32;
              default "15999";
              description
                "Upper value in the label range.";
          }
        }
```
The must statement ensure that upper bound is greater than lower bound. Type uint32 ensure that you could not enter a negative number.

And, I added `pre-validate` function for SRLB and SRGB to check that ranges are not overlap each other. Look at `isis_instance_segment_routing_srgb_pre_validate()` and `isis_instance_segment_routing_srlb_pre_validate()` functions in isis_nb_config.c file.

Reply:
The must statement ensure that upper bound is greater than lower bound. Type uint32 ensure that you could not enter a negative number.

And, I added pre-validate function for SRLB and SRGB to check that ranges are not overlap each other. Look at isis_instance_segment_routing_srgb_pre_validate() and isis_instance_segment_routing_srlb_pre_validate() functions in isis_nb_config.c file.